### PR TITLE
match_name() converts ald-sector to lowercase

### LINKED ISSUE #257: Maybe make `match_name` case-insensitive to input columns `sector` and `technology`
`match_name` is currently case sensitive to the input columns `sector` and `technology` (and probably others) of `ald_demo`. A bank has complained about this. 

I think it wouldn't hurt to be a little more flexible around this... although I'm not necessarily convinced. We would need to then make the rest of the r2dii ecosystem case insensitive, as this would likely probagate down to `r2dii.analysis` etc. Anyway, worth a discussion!

Thanks @georgeharris2deg 

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(r2dii.data)
library(r2dii.match)

loanbook_demo %>% 
  match_name(ald_demo)
#> # A tibble: 502 x 27
#>    id_loan id_direct_loant… name_direct_loa… id_intermediate… name_intermedia…
#>    <chr>   <chr>            <chr>            <chr>            <chr>           
#>  1 L1      C294             Yuamen Xinneng … <NA>             <NA>            
#>  2 L3      C292             Yuama Ethanol L… IP5              Yuama Inc.      
#>  3 L3      C292             Yuama Ethanol L… IP5              Yuama Inc.      
#>  4 L5      C305             Yukon Energy Co… <NA>             <NA>            
#>  5 L5      C305             Yukon Energy Co… <NA>             <NA>            
#>  6 L6      C304             Yukon Developme… <NA>             <NA>            
#>  7 L6      C304             Yukon Developme… <NA>             <NA>            
#>  8 L8      C303             Yueyang City Co… <NA>             <NA>            
#>  9 L9      C301             Yuedxiu Corp One IP10             Yuedxiu Group   
#> 10 L10     C302             Yuexi County AA… <NA>             <NA>            
#> # … with 492 more rows, and 22 more variables: id_ultimate_parent <chr>,
#> #   name_ultimate_parent <chr>, loan_size_outstanding <dbl>,
#> #   loan_size_outstanding_currency <chr>, loan_size_credit_limit <dbl>,
#> #   loan_size_credit_limit_currency <chr>, sector_classification_system <chr>,
#> #   sector_classification_input_type <chr>,
#> #   sector_classification_direct_loantaker <dbl>, fi_type <chr>,
#> #   flag_project_finance_loan <chr>, name_project <lgl>,
#> #   lei_direct_loantaker <lgl>, isin_direct_loantaker <lgl>, id_2dii <chr>,
#> #   level <chr>, sector <chr>, sector_ald <chr>, name <chr>, name_ald <chr>,
#> #   score <dbl>, source <chr>

loanbook_demo %>% 
  match_name(
    mutate(
      ald_demo, 
      sector = toupper(sector)
      )
  )
#> Warning: Found no match.
#> # A tibble: 0 x 27
#> # … with 27 variables: id_loan <chr>, id_direct_loantaker <chr>,
#> #   name_direct_loantaker <chr>, id_intermediate_parent_1 <chr>,
#> #   name_intermediate_parent_1 <chr>, id_ultimate_parent <chr>,
#> #   name_ultimate_parent <chr>, loan_size_outstanding <dbl>,
#> #   loan_size_outstanding_currency <chr>, loan_size_credit_limit <dbl>,
#> #   loan_size_credit_limit_currency <chr>, sector_classification_system <chr>,
#> #   sector_classification_input_type <chr>,
#> #   sector_classification_direct_loantaker <dbl>, fi_type <chr>,
#> #   flag_project_finance_loan <chr>, name_project <lgl>,
#> #   lei_direct_loantaker <lgl>, isin_direct_loantaker <lgl>, id_2dii <lgl>,
#> #   level <lgl>, sector <lgl>, sector_ald <lgl>, name <lgl>, name_ald <lgl>,
#> #   score <lgl>, source <lgl>
```

<sup>Created on 2020-08-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #257

### CODE REVIEW

File: tests/testthat/test-match_name.R [hunk 1]
```diff
@@ -571,3 +571,26 @@ test_that("outputs correct `borderline` (#269)", {
...
   expect_equal(out$borderline, c(FALSE, TRUE))
 })
+
+test_that("matches any case of ald$sector, but converts sector to lowercase", {
+  low <- match_name(fake_lbk(), fake_ald(sector = "power"))
+  expect_equal(low$sector, "power")
+
+  upp <- match_name(fake_lbk(), fake_ald(sector = "POWER"))
+  # The original uppercase is converted to lowercase
+  expect_equal(upp$sector, "power")
```
Comment (maurolepore) [inline tests/testthat/test-match_name.R:581]: @jdhoffa, is this okay?
i.e. with `ald$sector == "POWER" `match_name()` returns `sector_ald == "power".

File: tests/testthat/test-match_name.R [hunk 2]
```diff
@@ -571,3 +571,26 @@ test_that("outputs correct `borderline` (#269)", {
...
+
+test_that("matches any case of ald$name_company, but preserves original case", {
+  low <- match_name(fake_lbk(), fake_ald(name_company = "alpine knits"))
+  expect_equal(nrow(low), 1L)
+  expect_equal(low$name_ald, "alpine knits")
+
+  upp <- match_name(fake_lbk(), fake_ald(name_company = "ALPINE KNITS"))
+  expect_equal(nrow(upp), 1L)
+  # The original uppercase is preserved
+  expect_equal(upp$name_ald, "ALPINE KNITS")
```
Comment (maurolepore) [inline tests/testthat/test-match_name.R:595]: Notice the difference here: For `company_name` the case is preserved.

Comment (maurolepore) [reply tests/testthat/test-match_name.R:595]: Finally, #257 mentions `technology` but I don't see how this column is relevant to this issue. Can you show me a reprex?

``` r
devtools::load_all()
#> Loading r2dii.match

identical(
  match_name(fake_lbk(), fake_ald(technology = "a")),
  match_name(fake_lbk(), fake_ald(technology = "A"))
)
#> [1] TRUE
```

<sup>Created on 2020-08-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

File: tests/testthat/test-match_name.R (hunk 1 above)
Comment (jdhoffa) [reply tests/testthat/test-match_name.R:581]: yes, this is perfect.

File: tests/testthat/test-match_name.R (hunk 2 above)
Comment (jdhoffa) [reply tests/testthat/test-match_name.R:595]: Sorry, technology becomes relevant in `r2dii.analysis`, but not relevant here

**Comment by maurolepore** [comment]
Relates to 2degreesinvesting/r2dii.analysis#172

**Comment by jdhoffa** [comment]
@maurolepore happy to merge this now, unless you'd like to wait for 2DegreesInvesting/r2dii.analysis/pull/172?

**Pull request merged**